pyaedt/edb_grpc/core/components.py

```diff
@@ -0,0 +1,2209 @@
+            ]
+            for p in ref_pins:
+                if not p.IsLayoutPin():
+                    p.SetIsLayoutPin(True)
+            if len(ref_pins) == 0:
+                self._logger.info("No reference pin found on component {}.".format(component.GetName()))
+            if do_pingroup:
+                if len(ref_pins) == 1:
+                    self.create_terminal = self._create_terminal(ref_pins[0])
+                    self.terminal = self.create_terminal
+                    ref_pin_group_term = self.terminal
+                else:
+                    ref_pin_group = self.create_pingroup_from_pins(ref_pins)
+                    if not ref_pin_group:
+                        return False
+                    ref_pin_group_term = self._create_pin_group_terminal(ref_pin_group, component, isref=True)
+                    if not ref_pin_group_term:
+                        return False
+                for net in net_list:
+                    pins = [pin for pin in cmp_pins if pin.GetNet().GetName() == net]
+                    if pins:
+                        if len(pins) == 1:
+                            pin_term = self._create_terminal(pins[0])
+                            if pin_term:
+                                pin_term.SetReferenceTerminal(ref_pin_group_term)
+                        else:
+                            pin_group = self.create_pingroup_from_pins(pins)
+                            if not pin_group:
+                                return False
+                            pin_group_term = self._create_pin_group_terminal(pin_group, component)
+                            if pin_group_term:
+                                pin_group_term.SetReferenceTerminal(ref_pin_group_term)
+                    else:
+                        self._logger.info("No pins found on component {} for the net {}".format(component, net))
+            else:
+                ref_pin_group = self.create_pingroup_from_pins(ref_pins)
+                if not ref_pin_group:
+                    self._logger.warning("failed to create reference pin group")
+                    return False
+                ref_pin_group_term = self._create_pin_group_terminal(ref_pin_group, component, isref=True)
+                for net in net_list:
+                    pins = [pin for pin in cmp_pins if pin.GetNet().GetName() == net]
+                    for pin in pins:
+                        pin_group = self.create_pingroup_from_pins([pin])
+                        pin_group_term = self._create_pin_group_terminal(pin_group, component, isref=False)
+                        pin_group_term.SetReferenceTerminal(ref_pin_group_term)
+        return True
+
+    @pyaedt_function_handler()
+    def _create_terminal(self, pin):
+        """Create terminal on component pin.
+
+        Parameters
+        ----------
+        pin : Edb padstack instance.
+
+        Returns
+        -------
+        Edb terminal.
+        """
```

Comment:
```suggestion
        pin : EDB padstack instance.

        Returns
        -------
        EDB terminal.
        """
```
Data types are missing. Could the descriptions be more robust?